pkg/terraform/gather/azure/ip.go

```diff
@@ -11,17 +11,32 @@ import (
 )
 
 // BootstrapIP returns the ip address for bootstrap host.
-func BootstrapIP(tfs *terraform.State) (string, error) {
-	publicIP, err := terraform.LookupResource(tfs, "module.bootstrap", "azurerm_public_ip", "bootstrap_public_ip")
-	if err != nil {
-		return "", errors.Wrap(err, "failed to lookup public ip")
-	}
-	if len(publicIP.Instances) == 0 {
-		return "", errors.New("no public ip instance found")
-	}
-	bootstrap, _, err := unstructured.NestedString(publicIP.Instances[0].Attributes, "ip_address")
-	if err != nil {
-		return "", errors.New("no public_ip found for bootstrap")
+func BootstrapIP(tfs *terraform.State, private bool) (string, error) {
```

Comment:
i think you don't need the private, because if `bootstrap_public_ip` is not present, falling back to `azurerm_network_interface` should be fine to get bootstrap-host IP. we shouldn't need this explicit switch.

Reply:
I've updated it to work this way.